sdks/python/apache_beam/yaml/yaml_join_test.py

```diff
@@ -0,0 +1,210 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import unittest
+
+import apache_beam as beam
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+from apache_beam.yaml.yaml_transform import YamlTransform
+
+
+class ToRow(beam.PTransform):
+  def expand(self, pcoll):
+    return pcoll | beam.Map(lambda row: beam.Row(**row._asdict()))
+
+
+FRUITS = [
+    beam.Row(id=1, name='raspberry'),
+    beam.Row(id=2, name='blackberry'),
+]
+
+QUANTITIES = [
+    beam.Row(name='raspberry', quantity=1),
+    beam.Row(name='blackberry', quantity=2),
+    beam.Row(name='blueberry', quantity=3),
+]
+
+CATEGORIES = [
+    beam.Row(name='raspberry', category='juicy'),
+    beam.Row(name='blackberry', category='dry'),
+    beam.Row(name='blueberry', category='dry'),
+    beam.Row(name='blueberry', category='juicy'),
+]
+
+
+class YamlJoinTest(unittest.TestCase):
```

Comment:
Try adding the following annotation to skip this suite in pre-commits. It looks like the pre-commits do not have snapshot jars for the gradle providers.

```suggestion
@unittest.skipIf(
    TestPipeline().get_pipeline_options().view_as(StandardOptions).runner is
    None,
    'Do not run this test on precommit suites.')
class YamlJoinTest(unittest.TestCase):
```

Reply:
Have we verified these tests will actually get run if we're skipping them here? Should we add suite like https://github.com/apache/beam/blob/master/sdks/python/apache_beam/yaml/tests/csv.yaml ?

Reply:
@robertwb Have you verified that the integration_tests get run? I thought it ran pytest which would skip that file if it ends in 's', right? When I run pytest locally at least it doesn't run those

Reply:
It needs to be explicitly named, e.g.

`pytest -v apache_beam/yaml/integration_tests.py`

which is what

`./gradlew sdks:python:yamlIntegrationTests`

should invoke after building the jars. But this was just merged, so we should verify. (I don't think the `.runner is None` test passes in any context where the jars are built, but could be wrong.